### ISSUE TITLE
Probe: actor system

### ISSUE THREAD

**Issue by goodhoko** [tonarino/acoustic_profiler#5]
Explore implementing a probe into the actor framework.

idea: add USDT (dtrace) probes into actor framework and then use dtrace for the actual runtime tracing

**Comment by strohel** [comment]
Grabbing this one.

> idea: add USDT (dtrace) probes into actor framework and then use dtrace for the actual runtime tracing

Hmm, but wouldn't that make `actor` probing work only on MacOS?

But agreed making this suited for a particular library feels too specific, so I'll also look if I can easily hook in something more general, i.e.:
- #8
- [`tracing`](https://docs.rs/tracing/latest/tracing/) interface
- [`metrics`](https://docs.rs/metrics/latest/metrics/) interface

---
- This can end up being related to https://github.com/tonarino/actor/issues/7 and https://github.com/tonarino/actor/issues/27

**Comment by goodhoko** [comment]
> Hmm, but wouldn't that make actor probing work only on MacOS?

True. No one said it's a good idea 😅 (I just wanted to include a complete context.)
